errors.go

```diff
@@ -92,3 +114,8 @@ func IsServerError(err error) bool {
 func IsClientError(err error) bool {
 	return responseErrorType(err) == clientErr
 }
+
+// IsAuthError returns whether an error is authError.
+func IsAuthError(err error) bool {
```

Comment:
Should we write `IsAuthorizationError` so it's clear it's not about Authentication ?